pkg/schedule/schedule.go

```diff
@@ -149,17 +156,29 @@ func (f *fifo) run() {
 				f.mu.Unlock()
 				// clean up pending jobs
 				for _, todo := range pendings {
-					todo(f.ctx)
+					f.executeJob(todo, true)
 				}
 				return
 			}
 		} else {
-			todo(f.ctx)
+			f.executeJob(todo, false)
+		}
+	}
+}
+
+func (f *fifo) executeJob(todo Job, ctxDone bool) {
+	defer func() {
+		if !ctxDone {
 			f.finishCond.L.Lock()
 			f.finished++
 			f.pendings = f.pendings[1:]
 			f.finishCond.Broadcast()
 			f.finishCond.L.Unlock()
 		}
-	}
+		if err := recover(); err != nil {
+			f.lg.Warn("execute job failed", zap.Any("job", todo), zap.Any("err", err))
```

Comment:
What do you expect will be printed for `job` key? Just guessing that it will be pointer to function. In most cases we pass anonymous function to fifo. 

Maybe we could add a proper name to jobs scheduled? Could be a separate PR

Reply:
I want to give some tips when the panic happens. But I don’t have a good idea. Maybe deleting the job key is better and we add a proper name to jobs in a separate pr. What do you think of this?